spec/unit/lib/provider/securitydomain/post_wildfly_provider_spec.rb

```diff
@@ -0,0 +1,21 @@
+require "spec_helper"
+
+describe Puppet_X::Coi::Jboss::Provider::SecurityDomain::PostWildFlyProvider do
+  let(:resource) { {
+    :name          => 'testing',
+    :code          => 'Database',
+    :codeflag      => 'true',
+    :moduleoptions =>  {
+      'principalsQuery'   => 'select \'password\' from users u where u.login = ?',
+      'hashUserPassword'  => false,
+    },
+  } }
+
+  let(:provider) { double('mock', :resource => resource) }
+  let(:instance) { described_class.new(provider) }
+
+  describe '#create_parametrized_cmd with post wildfly' do
+    subject { instance.create_parametrized_cmd }
+    it { is_expected.to eq "subsystem=security/security-domain=testing/authentication=classic/login-module=UsersRoles:add(code=Database, flag=true,module-options=[(\"hashUserPassword\" => \"false\"),(\"principalsQuery\" => \"select 'password' from users u where u.login = ?\")]}])" }
```

Comment:
!resolve **Question:** I don't know that in this case is unneeded to invoke also: `subsystem=security/security-domain=testing/authentication=classic:add` before this. It could fail on real JBoss